app/custom-oauth/server/custom_oauth_server.js

```diff
@@ -290,6 +296,17 @@ export class CustomOAuth {
 		return username;
 	}
 
+	getAvatarUrl(data) {
+		const avatarUrl = this.avatarField.split('.').reduce(function(prev, curr) {
+			return prev ? prev[curr] : undefined;
+		}, data);
+		if (!avatarUrl) {
+			throw new Meteor.Error('field_not_found', `Avatar field "${ this.avatarField }" not found in data`, data);
```

Comment:
Should we prevent the login if avatar field is missing?

Reply:
That's a good question.. 

Maybe not.. Though the person configuring did say the avatar field would be there by setting the avatar field.  We do this with username field right above.

I personally would rather it fail like this on me like this so when i'm setting it up I know that I've not configured something exactly right

Reply:
I see your point, we might have an option to define if the username is required or not, but for username I can see some logic requiring it since if you are importing it from an external system you may have disabled the option to manual change it.

But avatars aren't too important to be in sync, and may even be more optional or less required by companies even when they want to import them when they exists.

What do you think?

Reply:
changed to debug 👍